cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpCredentialConnector.java

```diff
@@ -72,16 +66,148 @@ public CloudCredentialStatus create(AuthenticatedContext authenticatedContext) {
         throw new UnsupportedOperationException("Interactive login not supported on GCP");
     }
 
+    @Override
+    public CloudCredentialStatus delete(AuthenticatedContext authenticatedContext) {
+        Assert.notNull(authenticatedContext);
+        return new CloudCredentialStatus(authenticatedContext.getCloudCredential(), CredentialStatus.DELETED);
+    }
+
+    /**
+     * Creates a new CloudCredentialStatus instance with a FAILED status which
+     * contains the provided exception and some status reason about the cause
+     * of the status.
+     *
+     * @param e           The provided exception which stores the reason of
+     *                    the failed status.
+     * @param authContext The AuthenticatedContext instance which will be
+     *                    stored in the returning CloudCredentialStatus
+     *                    instance.
+     * @param message     The custom reason message which also be stored in the
+     *                    AuthenticatedContext.
+     *                    If the passed Optional is empty, then a default
+     *                    message is going to be passed to the status
+     *                    instance.
+     * @return The combined CloudCredentialStatus instance which stores all
+     * the necessary/required data for a proper object with a FAILED status.
+     */
+    private CloudCredentialStatus createFailedCloudCredentialStatusWithExc(Exception e, AuthenticatedContext authContext,
+            Optional<String> message) {
+        String logMessage = String.format("Could not verify credential, detailed message: %s", e.getMessage());
+        LOGGER.warn(logMessage, e);
+        return new CloudCredentialStatus(
+                authContext.getCloudCredential(),
+                CredentialStatus.FAILED,
+                e,
+                message.orElse("Could not verify credential!"));
+    }
+
+    /**
+     * Checks the validity of the provided GcpContext's credential. It sends
+     * a http request through the google's http api which validates the data
+     * set provided from the given GcpContext.
+     *
+     * @param gcpContext the GcpContext instance which credential would be
+     *                   checked.
+     * @throws IOException if something happens while listing the regions,
+     *                     this exception would thrown by the api.
+     */
     private void preCheckOfGooglePermission(GcpContext gcpContext) throws IOException {
+        /* The actual response which comes from the method invocation is ignored, because if
```

Comment:
Could you please remove documentations at all. Code is the best documentation, and we write documentation when the reason of the code is not clear somehow.